syncer/dml.go

```diff
@@ -419,14 +419,14 @@ func genKeyList(table string, columns []*model.ColumnInfo, dataSeq []interface{}
 			log.L().Debug("ignore null value", zap.String("column", columns[i].Name.O), zap.String("table", table))
 			continue // ignore `null` value.
 		}
-		// for key, I think no need to add the `,` separator.
+		// one column key looks like:`column_val.column_name`
 		buf.WriteString(columnValue(data, &columns[i].FieldType))
+		buf.WriteString(fmt.Sprintf(".%s.", columns[i].Name.String()))
```

Comment:
Sprintf is slower than +

Reply:
fixed in https://github.com/pingcap/dm/pull/2043/commits/3971a4dd2283c1facff14d208b0dcaff0ea9927e